opencensus_shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryCtx.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opencensusshim;
+
+import io.opencensus.trace.ContextHandle;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+
+public class OpenTelemetryCtx implements ContextHandle {
+
+  private final Context context;
+  private Scope scope;
+
+  public OpenTelemetryCtx(Context context) {
+    this.context = context;
+  }
+
+  Context getContext() {
+    return context;
+  }
+
+  @Override
+  public ContextHandle attach() {
+    scope = context.makeCurrent();
+    return this;
```

Comment:
From https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.html#attach--

> The previously current context is returned.

We may need to create a "child" context here somehow?

Reply:
I think we're okay 🙂  I've just added a [test](https://github.com/open-telemetry/opentelemetry-java/pull/1863/files#diff-c1ae03736614e7e31b5afcbb427c2eb673c78085cee664a1fd61ededab33a519R44) that tests parent-child relationships for:

- OpenTelemetry span (grandparent)
    - OpenCensus span (parent)
       - OpenTelemetry span (child)

Reply:
Thanks! Could you also add a test where all 3 spans are coming from OpenCensus? This is to make sure they don't overwrite each other.

Reply:
I think the use case is very small, but it's conceivable that user code relies on the return value being the previous value (they get values out of it expecting old values). So something like this could be more correct if I'm not mistaken.

```java
Context current = Context.current();
current.scope = this.context.makeCurrent();
return current;
```

Reply:
@nilebox yep, added

@anuraaga Did you mean something like this?
```
    // Get the current context, soon to be replaced
    Context prev = Context.current();
    // Replace the scope with the scope of `this` context
    scope = context.makeCurrent();
    // Return the previous context and the current scope
    context = prev;
    return this;
```
This would make the `context` inconsistent, as it would replace `this` context with the previous context every time `attach()` is called, whereas before `attach()` is called the context would be `this` context.

Reply:
Nah think my code is what I was thinking. The scope (used to close later) is attached to the previous context (which is what is normally returned by attach). Is there anything that would break by doing that?

Reply:
The `attach()` doesn't return a `Context`, it returns a `ContextHandle`, so to return the previous context I'd either have to do the above or create a new `ContextHandle` with the previous handle and the current scope — is this what you're suggesting?

Reply:
Ah yeah I think that's what I'm suggesting. I think it's required to match the definition of that API (return the previous context) but it could be reasonable to tell interop users this isn't supported because it's a very small use case for their to actually be an effect.